tests/e2e/magic_functions/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+# Set up C++ functions for e2e tests
+add_library(mg_procedure INTERFACE)
+target_include_directories(mg_procedure INTERFACE ${CMAKE_SOURCE_DIR}/include)
+
+function(add_query_module target_name src)
+  add_library(${target_name} SHARED ${src})
+  SET_TARGET_PROPERTIES(${target_name} PROPERTIES PREFIX "")
+  target_link_libraries(${target_name} PRIVATE mg_procedure)
+endfunction()
```

Comment:
As `add_library` would introduce a new global target, I would either remove that and put the `target_include_directories` in `add_query_module` or I would move this extra lib to the root `CMakeLists.txt` and name it `mg-procedure` (dash instead of underscore).